core/src/main/java/com/spotify/metrics/core/codahale/metrics/ext/Distribution.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright (c) 2016 Spotify AB.
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.spotify.metrics.core.codahale.metrics.ext;
+
+import com.codahale.metrics.Counting;
+import com.codahale.metrics.Metric;
+
+import java.nio.ByteBuffer;
+
+public interface Distribution extends Metric, Counting {
+
+    /**
+     * Record value from Min.Double to Max.Double.
+     *
+     * @param val
+     */
+    void record(double val);
+
+    /**
+     * Return serialized distribution and flush.
```

Comment:
what does flushing mean in this context? Does it wipe out the Distribution or just ensure it's properly serialised?

Reply:
Wipe distribution and  reset every state.